src/context.rs

```diff
@@ -323,7 +352,31 @@ impl<'a, 'b> ExtJsonBody for BodyReader<'a, 'b> {
 impl<'a, 'b> Read for BodyReader<'a, 'b> {
     ///Read the request body.
     fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
-        self.request.read(buf)
+        self.reader.read(buf)
+    }
+}
+
+///A specialized request representation for the multipart interface.
+pub struct MultipartRequest<'a> {
+    is_multipart: bool,
+    boundary: Option<&'a str>,
+    reader: &'a mut Read
```

Comment:
Woah, I missed this earlier. I put a lot of work into avoiding trait objects in I/O in `multipart`. `Read::read()` can be called frequently and means a trip through the vtable every time for a trait object.